local-cluster/src/Hydra/LogFilter.hs

```diff
@@ -0,0 +1,82 @@
+        Just "NewTx" -> replaceTransactionWithTxId node clientEvent
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkMessage
+        _ -> pure node
+    Just "ProcessingEffect" -> do
+      case node ^? clientEffect . key "tag" <|> node ^? networkEffect . key "tag" of
+        Just "SnapshotConfirmed" -> processSnapshotConfirmed node
+        Just "Committed" -> replaceUtxoWithNull node
+        Just "TxSeen" -> replaceTransactionWithTxId node clientEffect
+        Just "TxInvalid" -> replaceTransactionWithTxId node clientEffect >>= replaceUtxoWithNull
+        Just "TxValid" -> replaceTransactionWithTxId node clientEffect
+        Just "ReqTx" -> replaceTransactionWithTxId node networkEffect
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkEffect
+        _ -> pure node
+    Just "ProcessedEffect" -> do
+      case node ^? clientEffect . key "tag" <|> node ^? networkEffect . key "tag" of
+        Just "SnapshotConfirmed" -> processSnapshotConfirmed node
+        Just "Committed" -> replaceUtxoWithNull node
+        Just "TxSeen" -> replaceTransactionWithTxId node clientEffect
+        Just "TxInvalid" -> replaceTransactionWithTxId node clientEffect >>= replaceUtxoWithNull
+        Just "TxValid" -> replaceTransactionWithTxId node clientEffect
+        Just "ReqTx" -> replaceTransactionWithTxId node networkEffect
+        Just "ReqSn" -> replaceTransactionsWithTxIds node networkEffect
+        _ -> pure node
+    _ -> pure node
+  pure $ entry & _Object . at "message" ?~ result
+ where
+  processReqTx node = do
+    txid <- node ^? networkMessage . key "transaction" . key "id"
+    pure $ node & networkMessage . key "transaction" .~ txid
+
+  replaceTransactionsWithTxIds :: Value -> Traversal' Value Value -> Maybe Value
+  replaceTransactionsWithTxIds node path = do
+    let txids = node ^.. path . key "transactions" . values . key "id"
+    pure $ node & path . key "transactions" .~ Array (fromList txids)
+
+  processSnapshotConfirmed node = do
+    let txids = node ^.. clientEffect . key "snapshot" . key "confirmedTransactions" . values . key "id"
+    pure $
+      node
+        & clientEffect . key "snapshot" . key "confirmedTransactions" .~ Array (fromList txids)
+        & clientEffect . key "snapshot" . key "utxo" .~ Null
+
+  replaceUtxoWithNull node = pure $ node & clientEffect . key "utxo" .~ Null
+
+  replaceTransactionWithTxId :: Value -> Traversal' Value Value -> Maybe Value
+  replaceTransactionWithTxId node path = do
+    txid <- node ^? path . key "transaction" . key "id"
+    pure $ node & path . key "transaction" .~ txid
+
+  networkMessage :: Traversal' Value Value
+  networkMessage = key "event" . key "message"
+
+  networkEffect :: Traversal' Value Value
+  networkEffect = key "effect" . key "message"
+
+  clientEvent :: Traversal' Value Value
+  clientEvent = key "event" . key "clientInput"
+
+  clientEffect :: Traversal' Value Value
+  clientEffect = key "effect" . key "serverOutput"
```

Comment:
I wonder about the maintainability of all this though, when we change fields and whatnot; Since it's all JSON-based, we don't get any compiler warning / errors in case we alter the JSON logs and render these traversals invalid.